src/zenml/integrations/aws/service_connectors/aws_service_connector.py

```diff
@@ -0,0 +1,1877 @@
+            description="""
+Allows users to access one or more ECR repositories as a standard Docker
+registry resource. When used by connector consumers, they are provided a
+pre-authenticated python-docker client instance.
+
+The configured credentials must have at least the following ECR permissions for
+one or more ECR repositories:
+
+- ecr:DescribeRegistry
+- ecr:DescribeRepositories
+- ecr:ListRepositories
+- ecr:BatchGetImage
+- ecr:DescribeImages
+- ecr:BatchCheckLayerAvailability
+- ecr:GetDownloadUrlForLayer
+- ecr:InitiateLayerUpload
+- ecr:UploadLayerPart
+- ecr:CompleteLayerUpload
+- ecr:PutImage
+- ecr:GetAuthorizationToken
+
+This resource type is not scoped to a single ECR repository. Instead,
+a connector configured with this resource type will grant access to all the
+ECR repositories that the credentials are allowed to access under the configured
+AWS region (i.e. all repositories under the Docker registry URL
+`<account-id>.dkr.ecr.<region>.amazonaws.com`).
+
+The resource name associated with this resource type uniquely identifies an ECR
+registry using one of the following formats (the repository name is ignored,
+only the registry URL/ARN is used):
+            
+- ECR repository URI: https://<account-id>.dkr.ecr.<region>.amazonaws.com[/<repository-name>]
+- ECR repository ARN: arn:aws:ecr:<region>:<account-id>:repository[/<repository-name>]
+
+ECR repository names are region scoped. The connector can only be used to access
+ECR repositories in the AWS region that it is configured to use.
+""",
+            auth_methods=AWSAuthenticationMethods.values(),
+            # Does not support instances, given that the connector
+            # provides access to all permitted ECR repositories under the
+            # same ECR registry in the configured AWS region.
+            supports_instances=False,
+            logo_url="https://public-flavor-logos.s3.eu-central-1.amazonaws.com/container_registry/docker.png",
+            emoji=":whale:",
+        ),
+    ],
+)
+
+
+class AWSServiceConnector(ServiceConnector):
+    """AWS service connector."""
+
+    config: AWSBaseConfig
+
+    _account_id: Optional[str] = None
+    _session_cache: Dict[
+        Tuple[str, Optional[str], Optional[str]],
+        Tuple[boto3.Session, Optional[datetime.datetime]],
+    ] = {}
+
+    @classmethod
+    def _get_connector_type(cls) -> ServiceConnectorTypeModel:
+        """Get the service connector type specification.
+
+        Returns:
+            The service connector type specification.
+        """
+        return AWS_SERVICE_CONNECTOR_TYPE_SPEC
+
+    @property
+    def account_id(self) -> str:
+        """Get the AWS account ID.
+
+        Returns:
+            The AWS account ID.
+
+        Raises:
+            AuthorizationException: If the AWS account ID could not be
+                determined.
+        """
+        if self._account_id is None:
+
+            logger.debug("Getting ECR registry ID from AWS...")
```

Comment:
Is this really the ECR registry ID that we're getting here? Same for the error message below